plugins/rebalance/rest.go

```diff
@@ -0,0 +1,341 @@
+package rebalance
+
+import (
+	"net/http"
+
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/pkg/errors"
+
+	rebalanceapi "github.com/gluster/glusterd2/plugins/rebalance/api"
+
+	"github.com/gorilla/mux"
+	"github.com/pborman/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+func createRebalanceInfo(volname string, req *rebalanceapi.StartReq) *rebalanceapi.RebalInfo {
+	return &rebalanceapi.RebalInfo{
+		Volname:     volname,
+		RebalanceID: uuid.NewRandom(),
+		Status:      rebalanceapi.Started,
+		Cmd:         getCmd(req),
+		CommitHash:  setCommitHash(),
+		RebalStats:  []rebalanceapi.RebalNodeStatus{},
+	}
+}
+
+func rebalanceStartHandler(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	// collect inputs from url
+	volname := mux.Vars(r)["volname"]
+
+	var req rebalanceapi.StartReq
+
+	//  Unmarshal Request to check for fix-layout and start force
+
+	err := restutils.UnmarshalRequest(r, &req)
+
```

Comment:
missing error handling?

Reply:
I want to ignore the error in case this was not provided in the request. Not sure if this is the best way to do it but it works for now.

Reply:
Unmarshal only fails when bad request format. Error handling is required here. Default values will be available for later use if user not specified.

Reply:
If I check for an error, rebalance start fails if I do not provide either option.
[root@rhgs313-6 src]# curl -X POST http://192.168.122.6:24007/v1/volumes/gd2vol/rebalance/start
{"errors":[{"code":1,"message":"EOF"}]}